### ISSUE TITLE
Treat cancelled Apple subscriptions as invalid

### ISSUE THREAD

**Issue by cdriehuys** [knowmetools/km-api#474]
**What problem does your proposed feature solve?**
Currently if an Apple subscription is cancelled by Apple support, our app still treats it as valid until the original expiration date.

**Describe the solution you'd like**
If the most recent receipt transaction has the `cancellation_date` field set, it should be treated as invalid. This needs to be checked both when uploading the receipt and when attempting to renew it.

**Comment by cdriehuys** [comment]
Instead of deactivating cancelled subscriptions, we should delete them. The rationale stems from the following situation.

### TL;DR

> Subscriptions cancelled by Apple support should be deleted so the user can re-subscribe on another Know Me account.

### Cancellation Scenario

If a user loses access to their account for some reason but has an active subscription for that account, they have no way of stopping their subscription through the app. Instead they must either let the subscription lapse or cancel it immediately through Apple's customer support. The user's choice here indicates some sort of preference. If they choose to let the subscription lapse, we assume that they no longer need the premium status, and they do not intend to subscribe from another account.

On the other hand, if they cancel their subscription through Apple support, that indicates they intend to resubscribe on a new account that they have access to (barring any cancellation for refund reasons or similar). This poses an issue relating to the current behavior in that the user may not add their subscription to a new Know Me account while their old account is still using the same Apple receipt, even if it inactive.

### Notes

The above scenario is dependent on Apple keeping the same original transaction ID even if the subscription is cancelled and then repurchased.
